packages/browser/src/ClientAuthentication.ts

```diff
@@ -143,10 +143,17 @@ export default class ClientAuthentication {
   };
 
   handleIncomingRedirect = async (
-    url: string
+    url: string,
+    onError?: (
+      error: string | null,
+      errorDescription?: string | null
+    ) => unknown
   ): Promise<ISessionInfo | undefined> => {
-    const redirectInfo = await this.redirectHandler.handle(url);
-
+    const redirectInfo = await this.redirectHandler.handle(
+      url,
+      undefined,
+      onError
+    );
```

Comment:
That's where the issue was: it used to be 
```
    const redirectInfo = await this.redirectHandler.handle(
      url,
      onError
    );
```, which means the `onError` callback was passed as the `onNewToken` callback. I think we should consider changing the method signature to `(url: string, options: { onError: ..., onNewToken: ...})` to avoid confusion in the future (that could be done in a separate PR)